From 8598f1b2096f632eaa52d0cca49064b7ff72f4d6 Mon Sep 17 00:00:00 2001
From: Aihua Xu <aihuaxu@apache.org>
Date: Fri, 8 Jul 2016 09:54:12 -0400
Subject: [PATCH 638/723] CDH-42322 : HIVE-13749: Memory leak in Hive
 Metastore (Naveen Gangam, reviewed by Thejas M
 Nair, Aihua Xu)

(cherry picked from commit 935ec1b3e9197fb2af57789da5a27a5ec5be78f7)

Change-Id: I0e9e97fced6737c20e7665879ab38fdeae406964
(cherry picked from commit 3bf8ba8bd1f8111382674b90091dab1fce5cc969)
---
 .../hadoop/hive/metastore/HiveMetaStore.java       |    1 +
 1 file changed, 1 insertion(+)

diff --git a/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java b/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java
index 12d38b8..56dbc2b 100644
--- a/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java
+++ b/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java
@@ -876,6 +876,7 @@ public void shutdown() {
         try {
           ms.shutdown();
         } finally {
+          threadLocalConf.remove();
           threadLocalMS.remove();
         }
       }
-- 
1.7.9.5

